src/components/Toggle/SbToggle.vue

```diff
@@ -65,6 +71,11 @@ export default {
     handleInput(e) {
       this.$emit('update:modelValue', e.target.checked)
     },
+    handleClick() {
+      if (this.disabled) {
+        this.$emit('disabled-click')
+      }
+    },
```

Comment:
This is weird. I'm not sure @abenmassaoud Why do we need a handleClick method if we never emit the click itself but only the click when disabled. Can you explain better what's the purpose of this and why we are not also adding the emit of the click? Because if the toggle can be clicked when disabled I'm expecting it will also be when enable, isn't it?

Just looking at the code as it is, it's a little confusing..

Reply:
I confess that I also wondered about this Lori... 😬

Reply:
I guess because the wrapper still receives the click even if the input is disabled

Reply:
A question, In this case, shouldn't be better for the person who is using the `SbToogle` to add the 'click' listener, and make the logic based if it's disabled or not?

Reply:
Hi guys,

_> I guess because the wrapper still receives the click even if the input is disabled_

Exactly, when an HTML Element is `disabled`, the click event is not fired. And we have been asked to add a specific behavior if the user clicks on a disabled toggle. And this event was not fired, so I just added it 🙂

> _A question, In this case, shouldn't be better for the person who is using the SbToogle to add the 'click' listener, and make the logic based if it's disabled or not?_

This is what I wanted to do initially. But as I mentioned click event is not even fired when the element is disabled 🙂

Reply:
Oh I see 🤔 then maybe we could make the regular click event explicit out of the condition just for consistency?

Reply:
well, i made a test here and by writing this:

```
<div class="sb-toggle" :class="componentClasses" @click="click">
...
    click() {
      this.$emit('click')
    }
```

it works in both case, so maybe we can keep it simple and emit something like this: 

```
this.$emit('click', {disabled: this.disabled})
```

despite of the preventing conditional, what do u think @abenmassaoud @lisilinhart ?

Reply:
Fair enough 🙂 Done ✅ 

<img width="400" alt="Screenshot 2024-01-26 at 19 03 36" src="https://github.com/storyblok/storyblok-design-system/assets/16685155/754d82b8-193f-4764-b7d7-4b0ec8a165ca">